src/include/migraphx/bit_cast.hpp

```diff
@@ -30,8 +30,12 @@
 #include <migraphx/requires.hpp>
 #include <migraphx/config.hpp>
 
+#if defined(__GNUC__) and !defined(__clang__)
+#define MIGRAPHX_CONST_FOLD(x) (x)
+#else
 // NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
 #define MIGRAPHX_CONST_FOLD(x) (__builtin_constant_p(x) ? (x) : (x))
+#endif
```

Comment:
This is not correct. This needs to be defined for gcc. Only the warning should be disabled for the line below where its used.